MSAL/test/unit/MSALAdfsAuthorityResolverTests.m

```diff
@@ -0,0 +1,494 @@
+         XCTAssertNil(endpoint);
+         
+         XCTAssertTrue(error.code == MSALErrorFailedAuthorityValidation);
+         XCTAssertTrue([error.userInfo[MSALErrorDescriptionKey] containsString:@"DRS discovery"]);
+     }];
+
+}
+
+
+- (void)testOpenIDConfigurationEndpointForAuthority_whenDrsResponseMissingIdentityProviderService_shouldReturnError
+{
+    MSALAdfsAuthorityResolver *resolver = [MSALAdfsAuthorityResolver new];
+    NSURL *authority = [NSURL URLWithString:@"https://fs.fabrikam.com/adfs/"];
+    NSString *upn = @"displayable@contoso.com";
+    
+    [MSALTestSwizzle classMethod:@selector(queryEnrollmentServerEndpointForDomain:adfsType:context:completionBlock:)
+                           class:[MSALDrsDiscoveryRequest class]
+                           block:(id)^(id obj, NSString *domain, AdfsType type, id<MSALRequestContext> context, MSALDrsCompletionBlock completionBlock)
+     {
+         (void)domain;
+         (void)obj;
+         (void)type;
+         (void)context;
+         
+         MSALDrsDiscoveryResponse *response =
+         [[MSALDrsDiscoveryResponse alloc] initWithJson:@{} error:nil];
+         
+         completionBlock(response, nil);
+     }];
+    
+    
+    [resolver openIDConfigurationEndpointForAuthority:authority
+                                    userPrincipalName:upn
+                                             validate:YES
+                                              context:nil
+                                      completionBlock:^(NSString *endpoint, NSError *error)
+     {
+         XCTAssertNotNil(error);
+         XCTAssertNil(endpoint);
+         
+         XCTAssertTrue(error.code == MSALErrorFailedAuthorityValidation);
+         XCTAssertTrue([error.userInfo[MSALErrorDescriptionKey] containsString:@"DRS discovery"]);
+     }];
+    
+}
+
+- (void)testOpenIDConfigurationEndpointForAuthority_whenWebFingerResponseMissing_shouldReturnError
+{
+    MSALAdfsAuthorityResolver *resolver = [MSALAdfsAuthorityResolver new];
+    NSURL *authority = [NSURL URLWithString:@"https://fs.fabrikam.com/adfs/"];
+    NSString *upn = @"displayable@contoso.com";
+    
+    [MSALTestSwizzle classMethod:@selector(queryEnrollmentServerEndpointForDomain:adfsType:context:completionBlock:)
+                           class:[MSALDrsDiscoveryRequest class]
+                           block:(id)^(id obj, NSString *domain, AdfsType type, id<MSALRequestContext> context, MSALDrsCompletionBlock completionBlock)
+     {
+         (void)domain;
+         (void)obj;
+         (void)type;
+         (void)context;
+         
+         MSALDrsDiscoveryResponse *response =
+         [[MSALDrsDiscoveryResponse alloc] initWithJson:@{ @"IdentityProviderService" : @{ @"PassiveAuthEndpoint" : @"https://someendpoint.com"} } error:nil];
+         
+         completionBlock(response, nil);
+     }];
+    
+    
+    [MSALTestSwizzle classMethod:@selector(requestForAuthenticationEndpoint:authority:context:completionBlock:)
```

Comment:
These request/response tests were done on separate test classes, but I will see about restructuring some to have less swizzling here.